src/esmockLoader.js

```diff
@@ -81,6 +81,15 @@ const resolve = async (specifier, context, nextResolve) => {
 }
 
 const load = async (url, context, nextLoad) => {
+  if (url.endsWith('esmockIsLoader.js?test')) {
+    return {
+      format: 'module',
+      shortCircuit: true,
+      responseURL: encodeURI(url),
```

Comment:
you can hardcode responseURL anyhow starting with 'file://' (or even 'esmock://'? need to check, just for fun), nobody will use it.

I catch it due to encodeURI is a wrong function to encode something, we always must use encodeURIComponent to avoid stupid errors, see https://stackoverflow.com/questions/4540753/should-i-use-encodeuri-or-encodeuricomponent-for-encoding-urls) Sure, it's not a such case, but...

Reply:
We can try returning 'file://' or 'esmock://'. Regarding encodeURIComonent, esmock controls the url construction so that urls are compatible with encodeURI now and I prefer encodeURI currently.

Reply:
maybe just write responseURL: import.meta.url? it exists in fs, it valid, moreover we return shortCircuit: true which means that subsequent loaders will be bypassed.

Reply:
the url doesn't need to be encoded here this should be okay
```diff
-      responseURL: encodeURI(url),
+      responseURL: url,
```